runtime/src/comm/ugni/comm-ugni.c

```diff
@@ -5421,6 +5426,8 @@ void chpl_comm_execute_on(int locale, c_sublocid_t subloc,
            (int) locale, (int) subloc, (int) fid, arg, arg_size);
 
   if (locale == chpl_nodeID) {
+    assert(0); // locale model code should prevent this...
```

Comment:
Just seems better to get an error message for ease of reporting, but I don't feel strongly